app/src/org/commcare/preferences/DeveloperPreferences.java

```diff
@@ -378,4 +394,30 @@ public static boolean shouldShowUpdateOptionsSetting() {
                 CommCarePreferences.YES) || BuildConfig.DEBUG;
     }
 
+    private void hideDangerousDeveloperPrefsIfNeeded() {
+        if (!GlobalPrivilegesManager.isAdvancedSettingsAccessEnabled() && !BuildConfig.DEBUG) {
+            // Dangerous privileges should not be showing
+            PreferenceScreen prefScreen = getPreferenceScreen();
```

Comment:
Ah, thank you! I was trying to find a way to iterate through all prefs on the screen and was clearly looking in the wrong place.